compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertLoopsForVectorization.cpp

```diff
@@ -102,7 +156,7 @@ class AMDAIEInsertLoopsForVectorizationPass
       };
       auto lhsType = elType(genericOp->getOperand(0));
       auto rhsType = elType(genericOp->getOperand(1));
-      auto resType = elType(genericOp->getResult(0));
+      auto resType = elType(genericOp->getOperand(2));
```

Comment:
Was the former wrong? `auto resType = elType(genericOp->getResult(0));`. Why?

Reply:
Since this pass has been updated to work for `memref`, using `getResult(...)` doesn't work as the ops work only on operand basis. And this change I made would work for both `tensor` as well as `vector`.